File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-5.java
class Solution {
    public int findMin(int[] nums) {
        int left = 0;
        int right = nums.length - 1;

        while (left < right) {
            System.out.println("-------");
            int mid = left + (right - left) / 2;
            System.out.println("left = " + nums[left]);
            System.out.println("mid = " + nums[mid]);
            System.out.println("right = " + nums[right]);

            if(nums[mid] > nums[right]) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }

        return nums[left];
    }
}

File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-5.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:   
    def isSubtree(self, root: Optional[TreeNode], subRoot: Optional[TreeNode]) -> bool:
        
        if not root:
            return False

        if not subRoot:
            return True

        if self.sameTree(root, subRoot):
            return True
        
        return self.isSubtree(root.left, subRoot) or self.isSubtree(root.right, subRoot)

        return False
    
    def sameTree(self, node, subtreeNode):
        if not node and not subtreeNode:
            return True

        if node and subtreeNode and node.val == subtreeNode.val:
            return self.sameTree(node.left, subtreeNode.left) and self.sameTree(node.right, subtreeNode.right)

        return False